GoogleSignIn/Tests/Unit/GIDFakeMainBundle.m

```diff
@@ -37,16 +37,38 @@ @implementation GIDFakeMainBundle {
   NSMutableDictionary *_fakeConfig;
 }
 
-- (void)startFakingWithClientID:(NSString *)clientId {
-  _clientId = clientId;
+- (nullable instancetype)initWithClientID:(nullable id)clientID
+                           serverClientID:(nullable id)serverClientID
+                             hostedDomain:(nullable id)hostedDomain
+                              openIDRealm:(nullable id)openIDRealm {
+  self = [super init];
+
+  if (self) {
+    _clientId = clientID;
 
+    _fakeConfig = clientID ? [@{ @"GIDClientID" : clientID } mutableCopy] : [@{} mutableCopy];
+
+    [self fakeWithClientID:clientID
+            serverClientID:serverClientID
+              hostedDomain:hostedDomain
+               openIDRealm:openIDRealm];
+  }
+  return self;
+}
+
+- (instancetype)init {
```

Comment:
Please add to the `.h` and add docs that we pass nil to all of the designated initializer's parameters.